File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/session/Session.kt
// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package software.aws.toolkits.jetbrains.services.codemodernizer.session

data class Session(
    val tabId: String
) {
    var isAuthenticating: Boolean = false
    var authNeededNotified: Boolean = false
}

File: packages/amazonq/src/app/amazonqScan/chat/session/session.ts
/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

export enum ConversationState {
    IDLE,
    JOB_SUBMITTED,
}

export class Session {
    // Used to keep track of whether or not the current session is currently authenticating/needs authenticating
    public isAuthenticating: boolean = false

    // A tab may or may not be currently open
    public tabID: string | undefined

    public conversationState: ConversationState = ConversationState.IDLE

    constructor() {}

    public isTabOpen(): boolean {
        return this.tabID !== undefined
    }
}
